### ISSUE TITLE
Incorrect KMS resource validation when using shorthand SSM parameters

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#4193]
### CloudFormation Lint Version

1.38.0

### What operating system are you using?

Linux

### Describe the bug

[cfn-lint] I3510: action 'kms:Encrypt' requires a resource of ['arn:${Partition}:kms:${Region}:${Account}:key/.*']
[cfn-lint] I3510: action 'kms:Decrypt' requires a resource of ['arn:${Partition}:kms:${Region}:${Account}:key/.*']
[cfn-lint] I3510: action 'kms:GenerateDataKey' requires a resource of ['arn:${Partition}:kms:${Region}:${Account}:key/.*']

### Expected behavior

Preferably cfn-lint supports something along the lines of a parameter substitution json/yml file that we can set a value but in lieu of that, consistent behaviour between short-hand and parameter resolution.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'demo template'

Parameters:
  kmsKeyArn:
    Type: 'AWS::SSM::Parameter::Value<String>'
    Description: 'SSM Parameter Name containing kms key ARN. Default: /My/resources/kms_key_arn'
    Default: /My/resources/kms_key_arn

Resources:
  iAMPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties:
      ManagedPolicyName: 'DemoPolicy'
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:GenerateDataKey
            Resource: !Ref kmsKeyArn

  iAMPolicy2:
    Type: AWS::IAM::ManagedPolicy
    Properties:
      ManagedPolicyName: 'DemoPolicy'
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:GenerateDataKey
            Resource: "{{resolve:ssm:/My/resources/kms_key_arn}}"
```
